payment/payment.go

```diff
@@ -267,7 +275,7 @@ func (p *Payment) ConfidentialWitnessPubKeyHash() (string, error) {
 	}
 	addr, err := address.ToBlech32(payload)
 	if err != nil {
-		return "", nil
+		return "", err
```

Comment:
Nice catch! :)

Reply:
shall we make a dedicated PR and push right away?

Reply:
#196